core/azure-core/src/main/java/com/azure/core/polling/Poller.java

```diff
@@ -0,0 +1,269 @@
+        this.cancelOperation = cancelOperation;
+    }
+
+    /**
+     * This will call cancelOperation function if provided. Once cancelOperation is triggered successfully, it can not be called again.
+     * This is to avoid unintentional calls to cancelOperation.
+     * It will not call cancelOperation if operation status is not started/Cancelled/Failed/successfully completed.
+     *
+     * @throws UnsupportedOperationException when cancel operation is not provided.
+     */
+    public void cancelOperation() throws UnsupportedOperationException {
+        if (cancelOperation == null) {
+            throw new UnsupportedOperationException("Cancel operation is not supported on this service/resource.");
+        }
+
+        // We can not cancel an operation if it was never started
+        // or it is in its terminal state.
+        // Check cancelInitiated: to protect against multiple time call to cancel operation
+        if (cancelInitiated || (pollResponse != null && pollResponse.getStatus() != PollResponse.OperationStatus.IN_PROGRESS)) {
+            return;
+        }
+        //Time to call cancel
+        cancelOperation.accept(this);
+        cancelInitiated = true;
+    }
+
+    /**
+     * Enable user to subscribe and receive all the responses.
+     * The user will start receiving PollResponse When client subscribe to this Flux.
+     * This Flux will not trigger a call to PollOperation.
+     * The poller still have its own default polling in action unless, user has  turned off
+     * auto polling.
+     *
+     * @return poll response as Flux that can be subscribed.
+     */
+    public Flux<PollResponse<T>> getObserver() {
+        return fluxHandle;
+    }
+
+    /**
+     * Calls poll operation once in sync if Pool operation is not completed.
+     *
+     * @return a Mono of {@link PollResponse}
+     */
+    public Mono<PollResponse<T>> poll() {
+
+        if (!isTerminalState()) {
+            updatePollOperationAsync();
+        }
+        return Mono.just(pollResponse);
+    }
+
+    /**
+     * This will block till poll operation is complete
+     *
+     * @return returns poll response
+     */
+    public PollResponse<T> block() {
+        return fluxHandle.blockLast();
+    }
+
+    /*
+     * Calls poll operation function and update pollResponse.
+     */
+    private void updatePollOperationAsync() {
+        pollOperation.apply(pollResponse).subscribe(pResponse -> {
+            this.pollResponse = pResponse;
+        });
+    }
+
+    /*
+     * Get whether or not this PollStrategy's long running operation is done.
+     * @return Whether or not this PollStrategy's long running operation is done.
+     */
+    private Mono<PollResponse<T>> sendPollRequestWithDelay() {
+        return Mono.defer(() -> delayAsync().then(Mono.defer(() -> {
+            if (!isTerminalState()) {
+                updatePollOperationAsync();
+            }
+            return Mono.just(pollResponse);
```

Comment:
This code internally is equivalent to the poll() method. Should reduce duplication (once you've resolved how to handle the issue I pointed out in the poll method).

Reply:
resolved